src/test/unit/parameter_group.ld

```diff
@@ -0,0 +1,10 @@
+SECTIONS {
+  .pg_registry :   SUBALIGN(4)  /* force 4-byte alignment for pg_regisry. Gcc default is 32 bytes, causing padding insertion and breaking the init structure. */
+  {
+    PROVIDE_HIDDEN (__pg_registry_start = .);
+    KEEP (*(.pg_registry))
+    KEEP (*(SORT(.pg_registry.*)))
+    PROVIDE_HIDDEN (__pg_registry_end = .);
+  }
+}
+INSERT AFTER .fini_array;
```

Comment:
Can you post implicit `ld` script? just pass `-Wl,-verbose` to `g++` in linker step